catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/federation/impl/SortedFederationStrategy.java

```diff
@@ -274,9 +181,9 @@ private QueryResponse sourceFederate(List<Source> sources, final QueryRequest qu
           LOGGER.info("Plugin stopped processing", e);
         }
 
+        QueryRequest finalSourceQueryRequest = sourceQueryRequest;
         futures.put(
-            queryCompletion.submit(
-                new CallableSourceResponse(new TimedSource(source), sourceQueryRequest)),
+            queryCompletion.submit(() -> new TimedSource(source).query(finalSourceQueryRequest)),
```

Comment:
I think the new lambda could not access that local variable since it was not effectively final due to the `PreFederatedQueryPlugin`s could change the reference of that variable.  `finalSourceQueryRequest` is effectively final since it is never modified for the rest of its scope.